pcsx2/GameDatabase.cpp

```diff
@@ -182,68 +235,129 @@ GameDatabaseSchema::GameEntry YamlGameDatabaseImpl::entryFromYaml(const std::str
+		const ryml::substr view = c4::basic_substring<char>(buf.data(), size);
+		ryml::Tree tree = ryml::parse(view);
+		ryml::NodeRef root = tree.rootref();
+
+		for (const ryml::NodeRef& n : root.children())
 		{
-			// we don't want to throw away the entire GameDB file if a single entry is made incorrectly,
-			// but we do want to yell about it so it can be corrected
-			try
+			auto serial = StringUtil::toLower(std::string(n.key().str, n.key().len));
+
+			// Serials and CRCs must be inserted as lower-case, as that is how they are retrieved
+			// this is because the application may pass a lowercase CRC or serial along
+			//
+			// However, YAML's keys are as expected case-sensitive, so we have to explicitly do our own duplicate checking
+			if (gameDb.count(serial) == 1)
 			{
-				// Serials and CRCs must be inserted as lower-case, as that is how they are retrieved
-				// this is because the application may pass a lowercase CRC or serial along
-				//
-				// However, YAML's keys are as expected case-sensitive, so we have to explicitly do our own duplicate checking
-				std::string serial = strToLower(entry.first.as<std::string>());
-				if (gameDb.count(serial) == 1)
-				{
-					Console.Error(fmt::format("[GameDB] Duplicate serial '{}' found in GameDB. Skipping, Serials are case-insensitive!", serial));
-					continue;
-				}
-				gameDb[serial] = entryFromYaml(serial, entry.second);
+				Console.Error(fmt::format("[GameDB] Duplicate serial '{}' found in GameDB. Skipping, Serials are case-insensitive!", serial));
+				continue;
 			}
-			catch (const YAML::RepresentationException& e)
+			if (n.is_map())
 			{
-				Console.Error(fmt::format("[GameDB] Invalid GameDB syntax detected. Error Details - {}", e.msg));
+				gameDb[serial] = entryFromYaml(serial, n);
 			}
 		}
 	}
 	catch (const std::exception& e)
 	{
 		Console.Error(fmt::format("[GameDB] Error occured when initializing GameDB: {}", e.what()));
-		return false;
+		return;
 	}
+	ryml::set_callbacks(preserve_callbacks);
+	c4::set_error_callback(preserve_c4_callback);
+}
 
-	return true;
+const GameDatabaseSchema::GameEntry* YamlGameDatabaseImpl::findGame(const std::string& serial)
+{
+	// Lazy load the database if it hasn't been already
+	std::call_once(db_loaded, [this]() {
+		auto begin = std::chrono::steady_clock::now();
+		Console.WriteLn(fmt::format("[GameDB] Has not been initialized yet, initializing..."));
+		initDatabase();
+		auto end = std::chrono::steady_clock::now();
+		auto duration = std::chrono::duration_cast<std::chrono::milliseconds>(end - begin).count();
+		Console.WriteLn(fmt::format("[GameDB] Initialized in {}ms! {} games on record.", duration, gameDb.size()));
+	});
+
+	std::string serialLower = StringUtil::toLower(serial);
+	Console.WriteLn(fmt::format("[GameDB] Searching for '{}' in GameDB", serialLower));
+	if (gameDb.count(serialLower) == 1)
+	{
+		Console.WriteLn(fmt::format("[GameDB] Found '{}' in GameDB", serialLower));
+		return &gameDb[serialLower];
+	}
```

Comment:
If you refactor it to use `gameDb.find` and check the result instead, you'll get just one map lookup instead of two.